data_collection/web/guides.rst

```diff
@@ -10,3 +10,164 @@ Downloads and Outlinks
+
+.. _guide_tracking_category_and_product_views:
+
+Tracking category and product views
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Usually, the first e-commerce-related action a user performs on a website is browsing products. :ref:`setEcommerceView<js-api-setEcommerceView>` method allows us to track both category views and product views.
+
+To track a category view, use ``setEcommerceView`` method **before** tracking the page view, like this:
+
+.. code:: javascript
+
+    // set category to "Smartphones"
+    _paq.push(["setEcommerceView", false, false, "Smartphones"]);
+
+    // track page view
+    _paq.push(["trackPageView"]);
+
+The same method can be used for tracking product views. Again, it must be called **before** tracking a page view. Example:
+
+.. code:: javascript
+
+    // set product to "SUPER Phone A40 White"
+    _paq.push(["setEcommerceView", "71253029", "SUPER Phone A40 White", "Smartphones", 1499.99]);
+
+    // track page view
+    _paq.push(["trackPageView"]);
+
+The snippet above tracks product view of a smartphone with
+
+* SKU *71253029*
+* name *SUPER Phone A40 White*
+* category *Smartphones*
+* price *1499.99*
+
+``category`` parameter of the ``setEcommerceView`` method accepts not only string values, but also arrays of strings. This is useful for tracking products that belong to more than one category, or tracking pages that list products from multiple categories.
+
+.. code:: javascript
+
+    // set product to "SUPER Watch B20 Silver" belonging to 2 categories
+    _paq.push(["setEcommerceView", "00492710", "SUPER Watch B20 Silver", ["New offer", "Smartwatches"], 700.00]);
+
+    // track page view
+    _paq.push(["trackPageView"]);
+
+.. _guide_tracking_cart_updates:
+
+Tracking cart updates
+^^^^^^^^^^^^^^^^^^^^^
+
+Another type of e-commerce activity one can track is updating a shopping cart. With it, we are able to measure how often users don't complete the ordering process and what products stay in abandoned carts.
+
+Tracking a cart update has two steps: registering items from the cart and sending them. The following example uses two methods - :ref:`addEcommerceItem<js-api-addEcommerceItem>` and :ref:`trackEcommerceCartUpdate<js-api-trackecommercecartupdate>` - to achieve exactly that.
+
+.. code:: javascript
+
+    // user added one chocolate bar to an empty shopping cart
+
+    // register chocolate bar
+    _paq.push(["addEcommerceItem", "82775027", "MEGA Milk Chocolate 200g", "Candy", 6.00, 1]);
+
+    // track cart update
+    _paq.push(["trackEcommerceCartUpdate", 6.00]);
+
+This code snippet sends a cart update event with a cart containing one item (SKU *candy-12837*, name *MEGA Milk Chocolate 200g*, category *Candy*, price *6.00*) and having total value of *6.00*.
+
+The list of registered items **is not persisted** in the browser. If a user adds another item to the cart and the page reloads in between cart updates, the items that were already registered must be registered again.
```

Comment:
I'd just write that it's stateless. Every IT person should understand that.
It's important because it means that you have to send full content of the cart every time it's updated. It also means that of something is removed from the cart, then you have to send whole current content of the cart, just without this one item.
Maybe you should rephrase your description so it reflects the situation better.

```suggestion
The list of registered items **stateless**. If a visitor adds another item to the cart and the page reloads in between cart updates, the items that were already registered must be registered again.
```

Reply:
It's not completely stateless, it remembers what is in the cart until you reload a page, in case of a SPA that might be never. I'll try to formulate it differently.